### PR TITLE
fix(cli): warn about bundling updater target without appropriate targets, closes #7181

### LINKED ISSUE #7181: [bug] tauri build not create updater files
### Describe the bug

I followed the updater instruction in - https://tauri.app/v1/guides/distribution/updater/ 
in tauri.conf.json changed bundle.targets to "updater" 
and when I run tauri build got this:

   **Compiling updaterwork v0.0.0 (C:\Users\rivka.shaked\Downloads\UPDATERWORK\src-tauri)
    Finished release [optimized] target(s) in 20.56s
    Finished 1 bundle at:

        Info 0 updater archives at:**

no file created in \target\release\bundle

### Reproduction

[UPDATERWORK.zip](https://github.com/tauri-apps/tauri/files/11721499/UPDATERWORK.zip)

### Expected behavior

Updater package will be created with zip and sig file

### Platform and versions

```text
[✔] Environment
    - OS: Windows 10.0.19045 X64
    ✔ WebView2: 114.0.1823.43
    ✔ MSVC: Visual Studio Professional 2022
    ✔ rustc: 1.67.1 (d5a82bbd2 2023-02-07)
    ✔ Cargo: 1.67.1 (8ecd4f20a 2023-01-10)
    ✔ rustup: 1.25.2 (17db695f1 2023-02-01)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (default)
    - node: 18.14.0
    - npm: 9.3.1

[-] Packages
    - tauri [RUST]: 1.3.0
    - tauri-build [RUST]: 1.3.0
    - wry [RUST]: 0.24.3
    - tao [RUST]: 0.16.2
    - @tauri-apps/api [NPM]: not installed!
    - @tauri-apps/cli [NPM]: 1.3.1

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../src
    - devPath: ../src
```

### Stack trace

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by amrbashir**
### What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

**Comment by lucasfernog** [comment]
On Linux and macOS we're actually triggering the AppImage/app bundling. We just need to do the same on Windows.
https://github.com/tauri-apps/tauri/blob/696d77c3ce480f4a3b4c1c57dae64ff2bb7842ce/tooling/bundler/src/bundle/updater_bundle.rs#L121
https://github.com/tauri-apps/tauri/blob/696d77c3ce480f4a3b4c1c57dae64ff2bb7842ce/tooling/bundler/src/bundle/updater_bundle.rs#L78

**Comment by amrbashir** [comment]
I knew I saw this code somewhere lol.

Anyways, while this works fine for Linux and macOS, it is not the same case for Windows since there is **two** updater targets and not just **one**, so should we build both targets? doesn't sound intuitive to me, instead we should be disabling building `updater` target without another updater-enabled target, what do you think?

**Comment by lucasfernog** [comment]
What if we assume NSIS in this case? I think that should be our focus going forward, and maybe we can even drop WiX on v2 (probably move it out of the monorepo so custom users have time to migrate).

**Comment by amrbashir** [comment]
We can assume NSIS as default for now but it is not future-proof, if we decided to add another bundler target like `Squirrel` or maybe `MSIX`

**Comment by lucasfernog** [comment]
Hmm ok. Safer to go with this idea then.

**Review by lucasfernog — APPROVED** [review]

File: tooling/bundler/src/bundle.rs [hunk 1]
```diff
@@ -93,9 +97,23 @@ pub fn bundle_project(settings: Settings) -> crate::Result<Vec<Bundle>> {
...
+            PackageType::AppImage
+              | PackageType::MacOsBundle
+              | PackageType::Nsis
+              | PackageType::WindowsMsi
+          )
+        }) {
+          warn!("The updater bundle target exists but couldn't find any updater-enabled target, so the updater artifacts won't be generated. Please add one of these targets as well: app, appimage, msi, nsis");
+          continue;
+        }
+        updater_bundle::bundle_project(&settings, &bundles)?
```
Comment (AlexTMjugador) [inline tooling/bundler/src/bundle.rs:113]: I have just found out that `package_types` is never updated when a macOS DMG bundle is generated (see lines 77-85). Therefore, the updater bundle will not be generated if the macOS .app bundle is not explicitly set in `package_types`, even though the DMG bundle implies the generation of an app bundle and thus `package_types` does not need to explicitly contain a macOS .app bundle.

I'd suggest adding a `| PackageType::Dmg` branch to the `matches!` macro here to better handle this edge case.

**Pull request merged**